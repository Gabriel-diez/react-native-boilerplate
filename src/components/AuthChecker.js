/**
 * @flow
 */
import React, { Component } from 'react';
import {
    View,
    ActivityIndicator,
} from 'react-native';
import { Actions } from 'react-native-router-flux';
import firebase from 'firebase';

class AuthChecker extends Component {
    componentWillMount() {
        firebase.auth().onAuthStateChanged((user) => {
            if (user) {
                Actions.main({ type: 'reset' });
            } else {
                Actions.landing({ type: 'reset' });
            }
        });
    }

    render() {
        return (
            <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                <ActivityIndicator
                    size="large"
                />
            </View>
        );
    }
}

export default AuthChecker;
