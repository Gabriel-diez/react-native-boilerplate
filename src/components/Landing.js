import React from 'react';
import {
    Text,
    View,
    TouchableOpacity,
} from 'react-native';
import { Actions } from 'react-native-router-flux';

const Landing = () => {
    return (
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
            <Text>
                Landing Page
            </Text>

            <TouchableOpacity
                onPress={() => Actions.login()}
            >
                <Text>Connexion</Text>
            </TouchableOpacity>

            <TouchableOpacity
                onPress={() => Actions.register()}
            >
                <Text>Inscription</Text>
            </TouchableOpacity>
        </View>
    );
};

export default Landing;
