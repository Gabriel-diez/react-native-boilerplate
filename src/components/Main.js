import React from 'react';
import {
    Text,
    View,
} from 'react-native';

const Main = () => {
    return (
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
            <Text>
                Main App
            </Text>
        </View>
    );
};

export default Main;