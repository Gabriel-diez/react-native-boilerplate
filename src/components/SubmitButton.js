// @flow
import React from 'react';
import {
    Text,
    TouchableOpacity,
} from 'react-native';

type Props = {
    onPress: Object,
    buttonStyle?: Object,
    submitText?: string,
    submitTextStyle?: Object,
}

/**
 * Return <SubmitButton>
 * @param props
 * @returns {XML}
 * @constructor
 */
const SubmitButton = (props: Props) => {
    return (
        <TouchableOpacity
            style={{
                ...{
                    alignSelf: 'center',
                    backgroundColor: '#000',
                },
                ...props.buttonStyle,
            }}
            onPress={props.onPress}
        >
            <Text style={props.submitTextStyle}>
                {props.submitText}
            </Text>
        </TouchableOpacity>
    );
};

/**
 * @type {{submitText: string, submitTextStyle: {color: string}, buttonStyle: {paddingLeft: number, paddingRight: number, paddingTop: number, paddingBottom: number, borderRadius: number}}}
 */
SubmitButton.defaultProps = {
    submitText: 'Enregistrer',
    submitTextStyle: {
        color: '#FFF',
    },
    buttonStyle: {
        paddingLeft: 6,
        paddingRight: 6,
        paddingTop: 10,
        paddingBottom: 10,
        borderRadius: 5,
    },
};

export default SubmitButton;
