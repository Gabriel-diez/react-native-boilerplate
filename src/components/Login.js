import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
    View,
    Text,
    ActivityIndicator,
    TextInput,
} from 'react-native';
import {
    loginUser,
} from '../actions/index';
import SubmitButton from './SubmitButton';

const styles = {
    textInput: {
        height: 40,
        borderColor: 'grey',
        borderWidth: 1,
        marginTop: 10,
    },
};

export class Login extends Component {
    state = {
        email: '',
        password: '',
    };

    onEmailChange = (email) => {
        this.setState({
            email,
        });
    };

    onPasswordChange = (password) => {
        this.setState({
            password,
        });
    };

    onButtonPress = () => {
        const {
            email,
            password,
        } = this.state;

        this.props.loginUser({
            email,
            password,
        });
    };

    renderButton() {
        if (this.props.loading) {
            return (
                <View>
                    <ActivityIndicator />
                </View>
            );
        }

        return (
            <SubmitButton
                onPress={this.onButtonPress}
                submitText="Se connecter"
            />
        );
    }

    render() {
        return (
            <View style={{ flex: 1, justifyContent: 'center', margin: 10 }}>
                <Text style={{ alignSelf: 'center' }}>
                    Connexion
                </Text>
                <View>
                    <TextInput
                        style={styles.textInput}
                        onChangeText={this.onEmailChange}
                        placeholder="Email"
                        value={this.state.email}
                    />

                    <TextInput
                        style={styles.textInput}
                        secureTextEntry
                        placeholder="Password"
                        onChangeText={this.onPasswordChange}
                        value={this.state.password}
                    />
                </View>

                <Text>
                    {this.props.error}
                </Text>
                {this.renderButton()}
            </View>

        );
    }
}

const mapStateToProps = ({ auth }) => {
    const { error, loading } = auth;
    return { error, loading };
};

export default connect(mapStateToProps, {
    loginUser,
})(Login);
