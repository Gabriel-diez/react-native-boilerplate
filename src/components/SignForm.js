import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
    View,
    Text,
    ActivityIndicator,
    TextInput,
} from 'react-native';
import {
    loginUser,
    createUser,
} from '../actions/index';
import SubmitButton from './SubmitButton';

const styles = {
    container: {
        flex: 1,
        justifyContent: 'center',
        margin: 10,
    },
    textInput: {
        height: 40,
        borderColor: 'grey',
        borderWidth: 1,
        marginTop: 10,
    },
};

export class SignForm extends Component {
    static propTypes = {
        formType: PropTypes.oneOf(['login', 'register']).isRequired,
    };

    /**
     * initial state
     * @type {{email: string, password: string}}
     */
    state = {
        email: '',
        password: '',
    };

    onEmailChange = (email) => {
        this.setState({
            email,
        });
    };

    onPasswordChange = (password) => {
        this.setState({
            password,
        });
    };

    onButtonPress = () => {
        const {
            email,
            password,
        } = this.state;

        if (this.props.formType === 'login') {
            this.props.loginUser({
                email,
                password,
            });
        } else {
            this.props.createUser({
                email,
                password,
            });
        }
    };

    renderButton = () => {
        if (this.props.loading) {
            return (
                <View>
                    <ActivityIndicator
                        size="large"
                    />
                </View>
            );
        }

        return (
            <SubmitButton
                onPress={this.onButtonPress}
                submitText={this.props.formType === 'login' ? 'Se connecter' : 'S\'inscrire'}
            />
        );
    };

    renderTitle = () => {
        if (this.props.formType === 'login') {
            return 'Connexion';
        }
        return 'S\'enregistrer';
    };

    render() {
        return (
            <View style={styles.container}>
                <Text style={{ alignSelf: 'center' }}>
                    {this.renderTitle()}
                </Text>
                <View>
                    <TextInput
                        style={styles.textInput}
                        onChangeText={this.onEmailChange}
                        placeholder="Email"
                        value={this.state.email}
                    />

                    <TextInput
                        style={styles.textInput}
                        secureTextEntry
                        placeholder="Password"
                        onChangeText={this.onPasswordChange}
                        value={this.state.password}
                    />
                </View>

                <Text>
                    {this.props.error}
                </Text>
                {this.renderButton()}
            </View>

        );
    }
}

const mapStateToProps = ({ auth }) => {
    const { error, loading } = auth;
    return { error, loading };
};

export default connect(mapStateToProps, {
    loginUser,
    createUser,
})(SignForm);
