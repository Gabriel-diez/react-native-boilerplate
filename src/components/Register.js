import React from 'react';
import SignForm from './SignForm';

const Register = () => {
    return (
        <SignForm
            formType="register"
        />
    );
};

export default Register;

