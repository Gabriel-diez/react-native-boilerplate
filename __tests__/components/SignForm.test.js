import React from 'react';
import renderer from 'react-test-renderer';
import { SignForm } from '../../src/components/SignForm';

it('Sign Form renders correctly', () => {
    const snapshot = renderer.create(<SignForm />).toJSON();
    expect(snapshot).toMatchSnapshot();
});
