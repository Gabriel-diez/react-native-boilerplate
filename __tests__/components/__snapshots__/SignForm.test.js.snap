// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Form renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "margin": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    S'enregistrer
  </Text>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 40,
          "marginTop": 10,
        }
      }
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 40,
          "marginTop": 10,
        }
      }
      value=""
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#000",
        "borderRadius": 5,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
        }
      }
    >
      S'inscrire
    </Text>
  </View>
</View>
`;
